src/widgets/barchart/bar.rs

```diff
@@ -206,3 +230,52 @@ impl<'a> Bar<'a> {
         }
     }
 }
+
+impl<'a> From<(&'a str, u64)> for Bar<'a> {
+    fn from((label, value): (&'a str, u64)) -> Self {
+        Bar::new(label, value)
+    }
+}
+
+impl<'a> Styled for Bar<'a> {
+    type Item = Self;
+
+    fn style(&self) -> Style {
+        self.style
+    }
+
+    fn set_style<S: Into<Style>>(mut self, style: S) -> Self::Item {
+        self.style = style.into();
+        self
+    }
+}
```

Comment:
Can you give me a little helper blueprint on splitting this PR? I can reorganize the implementation and send separate PRs.

Reply:
How about these 3 PRs:
- `From<>` impl - breaking
- `Styled` impl
- new constructors and other things.

<details><summary>If your not super familiar with splitting up existing commits, at a high level the process is:</summary>


- create a new branch for the first PR 
- soft reset to the point where you branched from (there are 3 commits in this branch right now so that's `git reset HEAD~3`). This leaves the changes from the 3 commits in the index
- `git add --patch` the changed lines for the first commit
- commit that
- stash and test
- unstash
- cut a PR from that branch
- repeat

If in doubt, `git reflog` is your friend, and you should always be able to look at the hash of any commit you've written and update your current branch to that commit with `git reset --hard <somehash>`

</details>

Reply:
> From<> impl - breaking
> Styled impl
> new constructors and other things.

These sound good to me, I will split this pr according to this guideline! And thanks a lot for the tips about splitting existing commits, I have never done this before so this is helpful.

Out of topic, I have an internship interview soon and  I'm currently studying for that. After I take the interview, on November 5, I'll start working on this PR.

Reply:
No problem - take your time and good luck!

Reply:
Best of luck man!

Reply:
@joshka @orhun, thanks a lot for the good wishes :). It was an online assignment, and I decided to do it today, it went well. Let's see what happens next.

Back to work now, I assume you meant `Into<> impl` instead of `From<> impl` right?

Reply:
@joshka I've done the changes however, I think I didn't do it with the way you wanted. 

To prevent conflicts, I opened 3 different branches that are built on top of each other (`Into<> impl - breaking` ->
`Styled impl` -> `new constructors and other things`). To explain, they all contains one commit, however `Styled impl` has the changes from the first branch, `Into<> impl - breaking`. Will this cause a problem? If it's not, I will open the PRs. If the PRs are merged in the correct order, there will be no problem.  But if you want the PR's completely separated from each other, I can separate them completely and then send PR's and resolve the conflicts later.